### PR TITLE
FIX-#5492: Fix `Series.values` when `Series.dtype==ExtensionDtype`

### LINKED ISSUE #5492: `Series.values` works wrong when `Series.dtype=ExtensionDtype`
Initially found in https://github.com/modin-project/modin/pull/5469#discussion_r1052494972.

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5492 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/series.py [hunk 1]
```diff
@@ -486,6 +486,9 @@ def values(self):  # noqa: RT01, D200
         """
         import modin.pandas as pd
 
+        if isinstance(self.dtype, pandas.core.dtypes.dtypes.ExtensionDtype):
```
Comment (anmyachev) [inline modin/pandas/series.py:489]: Modin doesn't have core `module`.

File: modin/pandas/test/test_series.py [hunk 2]
```diff
@@ -3624,6 +3624,16 @@ def test_values_non_numeric():
     df_equals(modin_series.values, pandas_series.values)
 
 
+def test_values_ea():
+    data = pandas.arrays.SparseArray(np.arange(10, dtype="int64"))
```
Comment (anmyachev) [inline modin/pandas/test/test_series.py:3628]: Modin doesn't have `arrays` module.

File: modin/pandas/series.py (hunk 1 above)
Comment (jbrockmendel) [reply modin/pandas/series.py:489]: this should make the check on L493 unnecessary

Comment (anmyachev) [reply modin/pandas/series.py]: @jbrockmendel thanks! It was a mistake. Fixed.

**Review by dchigarev — APPROVED** [review]

**Pull request merged**